### DISCUSSION TITLE
Update Cloud Firestore to allow for not-in and !=

### DISCUSSION THREAD

**Discussion by TMcLoone** [invertase/react-native-firebase#4342] [Ideas] [answered]
[Firestore added a nice feature ](https://firebase.google.com/support/release-notes/js#version_7210_-_september_17_2020) that lets us search for fields that exits/don't and seems like react-native-firebase doesn't use the latest version. It would be awesome to have this as the current method is complicated especially if you have two where queries

**Comment by mikehardy** [comment, marked as answer]
I saw that as well! This would be great, we'd love a PR to implement it. It isn't even that hard I think - here is a template doing the same for the last batch of operators they added, so it can be followed as a guide: https://github.com/invertase/react-native-firebase/pull/2868/files

(that PR is mostly big because it moved the SDK dependencies up to the level that supports the new queries, the actual implementation isn't huge)

Note that to develop for react-native-firebase isn't too hard either - the "tests" folder has a README explaining how to get tests running, then you open Xcode or Android Studio on the "tests" directory as if it were an app you were developing, and the native code (plus javascript of course) is available through it's dependencies into the other modules. It ends up being pretty breezy, or at least as breezy as developing any react-native app is

**Reply by mikehardy** [reply]
Will be supported in #4474

**Comment by Salakar** [comment]
As a side note, both the Android & iOS Firebase SDKs will need to have this supported first before RNFirebase can, I'd be surprised if they both already support it, usually a couple weeks behind JS SDK.

**Reply by mikehardy** [reply]
They were quick on that one, with android release yesterday they both have it already:
https://firebase.google.com/support/release-notes/android#firestore_v21-7-0
https://firebase.google.com/support/release-notes/ios#cloud-firestore

**Reply by mikehardy** [reply]
Only testing complexity I think is to make sure this constraint is explicit in the API doc if we get a PR " Neither query operator will match documents where the specified field is not present."

**Reply by Salakar** [reply]
cc @russellwheatley may interest you if you want to implement a couple of new query features, for RNF & FF 🤓
